src/geo-brand-presence/handler.js

```diff
@@ -142,17 +152,32 @@ export async function keywordPromptsImportStep(context) {
     log,
   } = context;
 
+  let endDate;
+  let aiPlatform;
+
   /* c8 ignore start */
-  const endDate = Date.parse(data) ? data : undefined;
+  try {
+    // Try to parse as JSON first
+    const parsedData = JSON.parse(data);
+    if (parsedData.endDate && Date.parse(parsedData.endDate)) {
+      endDate = parsedData.endDate;
+    }
+    aiPlatform = parsedData.aiPlatform;
+  } catch (e) {
+    // If JSON parsing fails, treat as a date string (legacy behavior)
+    log.error('GEO BRAND PRESENCE:failed to parse %s as JSON: %j', data, e);
```

Comment:
```suggestion
    log.error('GEO BRAND PRESENCE:failed to parse %s as JSON:', data, e);
```
nit: don't add a placeholder for the error, coralogix and console.error will handle extra arguments fine and display them much more nicely, including a stack trace.